demos/examples/calcite/src/Components/LayerList/hooks.ts

```diff
@@ -2,25 +2,54 @@ import Handles from "@arcgis/core/core/Handles";
 import LayerListViewModel from "@arcgis/core/widgets/LayerList/LayerListViewModel";
 import { useEffect, useState } from "react";
 import { addLayerListHandlers } from "./utilities";
+import Collection from "@arcgis/core/core/Collection";
+import { CallBackData, LayerListViewModelState } from "./interfaces";
 
 export function useLayerListViewModel(
   props: __esri.LayerListViewModelProperties
 ) {
   const { view } = props;
   const [layerListVM, setLayerListVM] = useState<LayerListViewModel>(null);
-  const [count, setCount] = useState(0);
+  const [state, updateState] = useState<LayerListViewModelState>(null);
+  const [operationalItems, updateOperationalItems] = useState<
+    __esri.Collection<__esri.ListItem>
+  >(new Collection());
 
   useEffect(() => {
     const handles = new Handles();
 
     addLayerListHandlers({
       layerListVM,
       handles,
-      callback: () => {
-        // acceptable practice?
-        setCount((prev) => prev + 1);
-        console.log("COUNT: ", count);
-      },
+      callback: (data: CallBackData) => {
```

Comment:
Do you think it would be better to have multiple callbacks for the different data types instead of the if else statements? Just wondering if it might be cleaner to pass in multiple callbacks for each state.